frontend/src/lib/bare-metal-assets.ts

```diff
@@ -40,35 +40,7 @@ export async function syncBMAs(hosts: JsonArray, resources: JsonArray) {
         }
```

Comment:
There's a lot of code in this file that is not strongly typed.  When I load this branch locally, in vs code I see a lot of errors about types.  We are in the process of working on improving the linting; but we decided to use TypeScript and not JavaScript for a reason, so I would ask that we conform to the TypeScript way of doing things.

Reply:
fixed the typing